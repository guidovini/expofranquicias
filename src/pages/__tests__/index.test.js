import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { StaticQuery, useStaticQuery } from 'gatsby' // mocked

import IndexPage from '../index'

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: `Default Starter`,
        },
      },
    })
  )
  useStaticQuery.mockReturnValueOnce({
    site: {
      siteMetadata: {
        title: `Default Starter`,
        description: `Test description`,
        author: `Test author`,
      },
    },
  })
})

describe('Index Page', () => {
  test('Should_RenderIndexPageCorrectly', () => {
    const { context } = render(<IndexPage />)
    expect(context).toMatchSnapshot()
  })

  test('Should_HaveSearchFranchisesButton', () => {
    const { getByText } = render(<IndexPage />)
    expect(getByText('Buscar Franquicias')).toBeInTheDocument()
  })

  test('Should_HaveMoreInfoButton', () => {
    const { getByText } = render(<IndexPage />)
    expect(getByText('M치s Informaci칩n')).toBeInTheDocument()
  })

  // test('Should_RedirectPage_When_BuscarFranquiciasButtonCicked', async () => {
  //   const { getByText, unmount } = render(<IndexPage />)
  //   const searchFranchisesButton = getByText('Buscar Franquicias')
  //   fireEvent.click(searchFranchisesButton)
  //   expect(history.location.pathname).toBe('/franquicias-disponibles')
  // })

  // test('Should_RedirectPage_When_BuscarFranquiciasButtonCicked', () => {
  //   const { getByText } = render(<IndexPage />)
  //   const moreInfoButton = getByText('M치s Informaci칩n')
  //   fireEvent.click(moreInfoButton)
  //   expect(history.location.pathname).toBe('/contacto')
  // })
})
