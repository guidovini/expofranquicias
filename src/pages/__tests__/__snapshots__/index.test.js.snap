// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page Should_RenderIndexPageCorrectly 1`] = `undefined`;
